consensus/src/chained_bft/block_storage/block_store.rs

```diff
@@ -148,6 +151,62 @@ impl<T: Payload> BlockStore<T> {
         tree
     }
 
+    /// Commit the given block id with the proof, returns the path from current root or error
+    pub async fn commit(
+        &self,
+        block_id_to_commit: HashValue,
+        finality_proof: LedgerInfoWithSignatures,
```

Comment:
This PR didn't introduce this, but rather continues the pattern of passing both the block_id_to_commit and the finality_proof.  Can't we just derive the block_id_to_commit from finality_proof.ledger_info().consensus_block_id() and not pass in block_id_to_commit?